addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/avmfritz/internal/hardware/callbacks/FritzAhaUpdateXmlCallback.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.avmfritz.internal.hardware.callbacks;
+
+import java.io.StringReader;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.avmfritz.handler.IFritzHandler;
+import org.openhab.binding.avmfritz.internal.ahamodel.DeviceModel;
+import org.openhab.binding.avmfritz.internal.ahamodel.DevicelistModel;
+import org.openhab.binding.avmfritz.internal.hardware.FritzahaWebInterface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Callback implementation for updating multiple numbers decoded from a xml
+ * response. Supports reauthorization.
+ * 
+ * @author Robert Bausdorf
+ * 
+ */
+public class FritzAhaUpdateXmlCallback extends FritzAhaReauthCallback {
+	/**
+	 * logger
+	 */
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	/**
+	 * Handler to update
+	 */
+	private IFritzHandler handler;
+
+	/**
+	 * Constructor
+	 * @param webIface Webinterface to FRITZ!Box
+	 * @param handler Bridge handler taht will update things.
+	 */
+	public FritzAhaUpdateXmlCallback(FritzahaWebInterface webIface, IFritzHandler handler) {
+		super(WEBSERVICE_PATH, "switchcmd=getdevicelistinfos", webIface, Method.GET, 1);
+		this.handler = handler;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void execute(int status, String response) {
+		super.execute(status, response);
+		if (this.isValidRequest()) {
+			logger.trace("Received State response " + response);
+			try {
+				JAXBContext jaxbContext = JAXBContext
+						.newInstance(DevicelistModel.class);
+				Unmarshaller jaxbUM = jaxbContext.createUnmarshaller();
+
+				DevicelistModel model = (DevicelistModel) jaxbUM
+						.unmarshal(new StringReader(response));
+				for( DeviceModel device : model.getDevicelist() ) {

```

Comment:
you need to add a null check as model.getDevicelist() might be null (if there are no devices)